src/ir_def/construct.ml

```diff
@@ -76,7 +76,9 @@ let wildP =
 (* Primitives *)
 
 let varE (id, typ) =
```

Comment:
If we just add a flag to VarE, then varE would always construct a VarE, not one of two diffferent things.

Reply:
Correct, there is some charm to it!